core/src/components/action-sheet/action-sheet.tsx

```diff
@@ -230,6 +230,18 @@ export class ActionSheet implements ComponentInterface, OverlayInterface {
    */
   @Method()
   async dismiss(data?: any, role?: string): Promise<boolean> {
+    /**
+     * When using an inline action sheet
+     * and presenting an action sheet it is possible to
+     * quickly dismiss the action sheet while it is
+     * presenting. We need to await any current
+     * transition to allow the present to finish
+     * before dismissing again.
+     */
+    if (this.currentTransition !== undefined) {
+      await this.currentTransition;
```

Comment:
I'm not sure if I'm testing this correctly, but the fix does not appear to work.

In the `action-sheet/test/trigger` test I added the following to the script. I expect the action sheet to remain open, but it disappear after this code executes.

```js
setTimeout(() => {
  actionSheet.isOpen = true;
  setTimeout(() => {
    actionSheet.isOpen = false;
    setTimeout(() => {
      actionSheet.isOpen = true;
    }, 50);
  }, 50);
}, 2000);
```

https://github.com/ionic-team/ionic-framework/assets/2721089/e6fe9c16-8299-4c05-bc9b-4cc7c249c275